src/Compilers/CSharp/Test/Semantic/Semantics/NullableContextTests.cs

```diff
@@ -413,8 +430,1174 @@ void verify(CSharpParseOptions parseOptions, bool expectedFlowState, params stri
             }
         }
 
-        private static string[] GetNullableDataKeysAsStrings(ConcurrentDictionary<object, int> nullableData) =>
-            nullableData.Keys.Select(key => GetNullableDataKeyAsString(key)).OrderBy(key => key).ToArray();
+        private readonly struct NullableDirectives
+        {
+            internal readonly string[] Directives;
+            internal readonly NullableContextState.State ExpectedWarningsState;
+            internal readonly NullableContextState.State ExpectedAnnotationsState;
+
+            internal NullableDirectives(string[] directives, NullableContextState.State expectedWarningsState, NullableContextState.State expectedAnnotationsState)
+            {
+                Directives = directives;
+                ExpectedWarningsState = expectedWarningsState;
+                ExpectedAnnotationsState = expectedAnnotationsState;
+            }
+        }
+
+        [Fact]
+        [WorkItem(49746, "https://github.com/dotnet/roslyn/issues/49746")]
+        public void AnalyzeMethodsInEnabledContextOnly_01()
+        {
+            var projectSettings = new[]
+            {
+                (NullableContextOptions?)null,
+                NullableContextOptions.Disable,
+                NullableContextOptions.Warnings,
+                NullableContextOptions.Annotations,
+                NullableContextOptions.Enable,
+            };
+            var nullableDirectives = new[]
```

Comment:
Consider leaving a comment with the actual expected format of these directives. It's pretty hard to visualize as it right now.

Reply:
Just a single general comment with a line of `annotations go here` would suffice.

---
In reply to: [546881986](https://github.com/dotnet/roslyn/pull/49953#discussion_r546881986) [](ancestors = 546881986)

Reply:
Sorry, this was in reference to the final produced code.

---
In reply to: [546882235](https://github.com/dotnet/roslyn/pull/49953#discussion_r546882235) [](ancestors = 546882235,546881986)